### ISSUE TITLE
Unable to start and exit directly

### ISSUE THREAD

**Issue by arcosx** [shadowsocks/shadowsocks-rust#476]
### OS Version
4.4.0-206-generic
Linux wgbdev 4.4.0-206-generic #238-Ubuntu SMP Tue Mar 16 07:52:37 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
### Error Info
```bash
RUST_BACKTRACE=full sslocal -b "127.0.0.1:8765" --server-url "ss://fakeaddress"
thread 'main' panicked at 'Malformed input', /root/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/shadowsocks-1.10.0/src/config.rs:330:18
stack backtrace:
   0:     0x55b0d26127ba - std::backtrace_rs::backtrace::libunwind::trace::h4dee703919bfd40a
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x55b0d26127ba - std::backtrace_rs::backtrace::trace_unsynchronized::h457e839f1a563e20
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55b0d26127ba - std::sys_common::backtrace::_print_fmt::h86a55fb30f8393c8
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55b0d26127ba - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h7b3d6cac46d277e1
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55b0d241bfaf - core::fmt::write::h127419eb46f2ecc9
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/core/src/fmt/mod.rs:1092:17
   5:     0x55b0d2612094 - std::io::Write::write_fmt::h6010cfbb4726588b
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/io/mod.rs:1578:15
   6:     0x55b0d2611c15 - std::sys_common::backtrace::_print::h79b4f9652330cc9d
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55b0d2611c15 - std::sys_common::backtrace::print::h330bb326a76af8cf
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55b0d2611c15 - std::panicking::default_hook::{{closure}}::heb6a42a7d50a472e
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/panicking.rs:208:50
   9:     0x55b0d2611192 - std::panicking::default_hook::h17e521ba6d68d6e1
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/panicking.rs:225:9
  10:     0x55b0d2611192 - std::panicking::rust_panic_with_hook::h70db735e3a6e70cb
                               at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/panicking.rs:591:17
  11:     0x55b0d258e8b3 - std::panicking::begin_panic::{{closure}}::h3656ef9cd047e1e3
  12:     0x55b0d258e88c - std::sys_common::backtrace::__rust_end_short_backtrace::h88c853f8182f715c
  13:     0x55b0d225910a - std::panicking::begin_panic::h49873ef1fac6041f
  14:     0x55b0d2591580 - shadowsocks::config::ServerConfig::from_url::h58f00175aecc9406
  15:     0x55b0d22d2e77 - core::ops::function::Fn::call::h18f91bee8bc597ea
  16:     0x55b0d24084c3 - clap::app::validator::Validator::validate::h950ad6e889568247
  17:     0x55b0d2416b6a - clap::app::parser::Parser::get_matches_with::hadf21fe1288acb6a
  18:     0x55b0d23b3c77 - sslocal::main::hcd13baff71c5c1fd
  19:     0x55b0d22d0776 - std::sys_common::backtrace::__rust_begin_short_backtrace::h0f9682878f616040
  20:     0x55b0d23c3bf8 - main
  21:     0x7f0b56749840 - __libc_start_main
  22:     0x55b0d2260039 - _start
  23:                0x0 - <unknown>
Aborted (core dumped)
```

**Comment by DuckSoft** [comment]
复现成功，此处的错误处理确实似乎不够健壮……

**Comment by zonyitoo** [comment]
Should be fixed by `return Err(UrlParseError::InvalidUserInfo)`

**Comment by zonyitoo** [comment]
fixed by #478 .
